### DISCUSSION TITLE
gradle clean build报错

### DISCUSSION THREAD

**Discussion by visen-zheng** [Tencent/bk-ci#6335] [Q&A] [answered]
环境:
gradle 7.4.1

gradle.properties采用默认配置:
# devops-framework 版本
devopsBootVersion=0.0.4
# 版本仓库配置
MAVEN_REPO_URL=http://mirrors.cloud.tencent.com/nexus/repository/maven-public
MAVEN_REPO_DEPLOY_URL=
MAVEN_REPO_USERNAME=
MAVEN_REPO_PASSWORD=
MAVEN_REPO_SNAPSHOT_DEPLOY_URL=
MAVEN_REPO_SNAPSHOT_USERNAME=
MAVEN_REPO_SNAPSHOT_PASSWORD=

报错如下:
$ gradle clean build
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :buildSrc:compilePluginsBlocks FAILED
e: /extdisk/data/go/src/bk-ci/src/backend/ci/buildSrc/build/kotlin-dsl/plugins-blocks/extracted/task-deploy-to-maven.gradle.kts:32:5: Function invocation 'maven(...)' expected
e: /extdisk/data/go/src/bk-ci/src/backend/ci/buildSrc/build/kotlin-dsl/plugins-blocks/extracted/task-deploy-to-maven.gradle.kts:32:5: Unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public fun RepositoryHandler.maven(url: Any): MavenArtifactRepository defined in org.gradle.kotlin.dsl
public fun RepositoryHandler.maven(url: Any, action: MavenArtifactRepository.() -> Unit): MavenArtifactRepository defined in org.gradle.kotlin.dsl

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildSrc:compilePluginsBlocks'.
> Script compilation errors:
  
    Line 32:     maven
                 ^ Function invocation 'maven(...)' expected
  
    Line 32:     maven
                 ^ Unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
                     public fun RepositoryHandler.maven(url: Any): MavenArtifactRepository defined in org.gradle.kotlin.dsl
                     public fun RepositoryHandler.maven(url: Any, action: MavenArtifactRepository.() -> Unit): MavenArtifactRepository defined in org.gradle.kotlin.dsl
  
  2 errors

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1m 26s
3 actionable tasks: 1 executed, 2 from cache

请教有人遇到过这个问题么? 帮帮忙 谢谢啦.

**Comment by slarkzhu** [comment, marked as answer]
估计是jdk版本不对，文档要jdk1.8

**Reply by irwinsun** [reply]
JDK 8
Gradle 6.+

其他的还未做适配验证，未知
